### PR TITLE
Publish an annotation on new specimen creation

### PR DESCRIPTION
**Pull request by samleeflang**
Send auto-accepted annotation to kafka.
Decided to just use the annotation-processing-request. However, the annotation-processor needs information about who approved the annotation, so that's why I included the acceptingAgent and needed to make an additional object.
Also didn't want to crash/revert everything if this failed (it is not **that** important) we could add annotations in later if we have an issue, plus it is only a writeValueToString for the ObjectMapper, that should never be an issue...
So we just log the ID and cry about it.
If it happens, we need to write a small script which creates auto-accepted annotation for the once we missed.

hmm think that is it, updates will be much trickier and we also need to do media (create user stories for this)

also did some upgrades of the libs and some auto-formatting 

----------------------------
Added functionality for auto-accepted updates as well as most of it impacted this PR.
- SourceSystemName is now part of the object so can be used without the need of the SourceSystemComponent retrieving it.
- UpdatedDigitalSpecimen object now also contains the JSON Patch as we use it in multiple places 
- Processing service also ignores Modified timestamp (otherwise it will always see a difference)
- Improved message in CreateUpdateTombstone event so it reflects the type of change

### CODE REVIEW

File: src/main/java/eu/dissco/core/digitalspecimenprocessor/domain/AutoAcceptedAnnotation.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package eu.dissco.core.digitalspecimenprocessor.domain;
+
+import eu.dissco.core.digitalspecimenprocessor.schema.Agent;
+import eu.dissco.core.digitalspecimenprocessor.schema.AnnotationProcessingRequest;
+
+public record AutoAcceptedAnnotation(
+    Agent acceptingAgent,
```
[the remark this answers is not in the document]
Comment (southeo) [reply src/main/java/eu/dissco/core/digitalspecimenprocessor/domain/AutoAcceptedAnnotation.java:7]: I can't find acceptingAgent in the [annotation schema](https://schemas.dissco.tech/schemas/fdo-type/annotation/0.3.0/annotation.json). I remember discussing this term, did we not add it to schemas.tech?

**Review by southeo — COMMENTED** [review]
:cookie:

File: src/main/java/eu/dissco/core/digitalspecimenprocessor/domain/AutoAcceptedAnnotation.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (samleeflang) [reply src/main/java/eu/dissco/core/digitalspecimenprocessor/domain/AutoAcceptedAnnotation.java:7]: This is an internal object. The annotation processing service will need to know who approved the annotation for the mergingStatus fields. However, I did want to reuse the annotation-processing-request object and not create a new object. So it now have this internal object with some info about the agent which the annotation-processor can then use to fill in the rest of the object (PID, mergeStatus, etc...)

File: src/main/java/eu/dissco/core/digitalspecimenprocessor/service/AnnotationPublisherService.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+  private List<AnnotationProcessingRequest> convertJsonPatchToAnnotations(
+      DigitalSpecimenRecord digitalSpecimenRecord, JsonNode jsonNode)
+      throws JsonProcessingException {
+    var annotations = new ArrayList<AnnotationProcessingRequest>();
+    var sourceSystemID = digitalSpecimenRecord.digitalSpecimenWrapper().attributes()
+        .getOdsSourceSystemID();
+    var sourceSystemName = digitalSpecimenRecord.digitalSpecimenWrapper().attributes()
+        .getOdsSourceSystemName();
+    for (JsonNode action : jsonNode) {
```
[the remark this answers is not in the document]
Comment (southeo) [reply src/main/java/eu/dissco/core/digitalspecimenprocessor/service/AnnotationPublisherService.java:130]: do you need an `isArray()` check first?

File: src/main/java/eu/dissco/core/digitalspecimenprocessor/service/AnnotationPublisherService.java [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+      return action.get(VALUE).textValue();
+    } else {
+      return mapper.writeValueAsString(action.get(VALUE));
+    }
+  }
+
+  private OaHasSelector buildSpecimenSelector(JsonNode action) {
+    var pointer = action.get("path").asText();
+    var path = convertJsonPointToJsonPath(pointer);
+    if (action.get("path").asText().endsWith("/-")) {
```
Comment (southeo) [inline src/main/java/eu/dissco/core/digitalspecimenprocessor/service/AnnotationPublisherService.java:170]: I couldn't find anything on a jsonPointer ending with "/-", what does this imply?

File: src/main/java/eu/dissco/core/digitalspecimenprocessor/service/AnnotationPublisherService.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (samleeflang) [reply src/main/java/eu/dissco/core/digitalspecimenprocessor/service/AnnotationPublisherService.java:130]: Hmm we generate this jsonNode a bit earlier by the JsonPatch library. We could check all the components but as it gives a fixed output I made some assumptions here. Initially I tried using the JsonPatch object but this hides the JsonPatchOperation so I couldn't access them. So JsonNode it was but the lib does give gaurantees it is always this format.

File: src/main/java/eu/dissco/core/digitalspecimenprocessor/service/AnnotationPublisherService.java (hunk 3 above)
Comment (samleeflang) [reply src/main/java/eu/dissco/core/digitalspecimenprocessor/service/AnnotationPublisherService.java:170]: "If the "-" character is used to index the end of the array (see [[RFC6901](https://datatracker.ietf.org/doc/html/rfc6901)]), this has the effect of appending the value to the array." So basically it indicates that it wasn't to add an object to an array. In the test I used the example of an EntityRelationship.

**Review by southeo — APPROVED** [review]
:paintbrush:

**Pull request merged**